instrumentation/spring-boot-actuator-3.0.0/src/main/java/org/springframework/boot/actuate/endpoint/web/servlet/AbstractWebMvcEndpointHandlerMapping_Instrumentation.java

```diff
@@ -0,0 +1,44 @@
+/*
+ *
+ *  * Copyright 2024 New Relic Corporation. All rights reserved.
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+package org.springframework.boot.actuate.endpoint.web.servlet;
+
+import com.newrelic.api.agent.NewRelic;
+import com.newrelic.api.agent.Trace;
+import com.newrelic.api.agent.Transaction;
+import com.newrelic.api.agent.TransactionNamePriority;
+import com.newrelic.api.agent.weaver.MatchType;
+import com.newrelic.api.agent.weaver.Weave;
+import com.newrelic.api.agent.weaver.Weaver;
+import com.nr.agent.instrumentation.actuator.SpringActuatorUtils;
+import jakarta.servlet.http.HttpServletRequest;
+
+import java.util.Map;
+
+@Weave(type = MatchType.BaseClass, originalName = "org.springframework.boot.actuate.endpoint.web.servlet.AbstractWebMvcEndpointHandlerMapping")
+public class AbstractWebMvcEndpointHandlerMapping_Instrumentation {
+    @Weave(type = MatchType.ExactClass, originalName = "org.springframework.boot.actuate.endpoint.web.servlet.AbstractWebMvcEndpointHandlerMapping$OperationHandler")
+    private static final class OperationHandler_Instrumentation {
+        @Trace
+        Object handle(HttpServletRequest request, Map<String, String> body) {
+            if (SpringActuatorUtils.isActuatorEndpointNamingEnabled) {
+                Transaction transaction = NewRelic.getAgent().getTransaction();
+
+                if (transaction != null) {
```

Comment:
The API call will always return not null.
You could check the class type
```
if (! transaction instanceof NoOpTransaction)
```
Or you can use the agent bridge

```
Transaction transaction = AgentBridge.getAgent().getTransaction(false);
```

Reply:
Good catch